internal/services/order/order_rest.go

```diff
@@ -0,0 +1,290 @@
+package order
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"Tradeasy/internal/services/stock_exchange"
+	"time"
+)
+
+func UpdateBuyOrder(res *stock_exchange.OrderResponse) (err error) {
+	var p model.PendingOrders
+	var account model.TradingAccount
+
+	if res.Status == failed {
+		//Buy order Failed
+		if err = config.DB.Table("pending_orders").Where("order_id=?", res.OrderID).First(&p).Error; err != nil {
+			return err
+		}
+		p.Status = failed
+		if err = config.DB.Table("pending_orders").Where("order_id=?", res.OrderID).Updates(&p).Error; err != nil {
+			return err
+		}
+		if err = config.DB.Table("trading_account").Where("user_id=?", p.UserId).First(&account).Error; err != nil {
+			return err
+		}
+		account.Balance = account.Balance + int64(p.Quantity*p.OrderPrice)
+		if err = config.DB.Table("trading_account").Where("user_id=?", p.UserId).Updates(&account).Error; err != nil {
+			return err
+		}
+		if err = config.DB.Table("pending_orders").Where("order_id=?", res.OrderID).Delete(&p).Error; err != nil {
+			return err
+		}
+		return nil
+	} else if res.Status == completed {
+		//Buy order completed
+		if err = config.DB.Table("pending_orders").Where("order_id=?", res.OrderID).First(&p).Error; err != nil {
+			return err
+		}
+		p.Status = completed
+		if err = config.DB.Table("pending_orders").Where("order_id=?", res.OrderID).Updates(&p).Error; err != nil {
+			return err
+		}
+		h := model.Holdings{
+			UserId:    p.UserId,
+			OrderId:   p.OrderId,
+			StockName: p.StockName,
+			Quantity:  p.Quantity,
+			BuyPrice:  res.AveragePrice,
+			OrderedAt: res.OrderExecutionTime,
+		}
+		if p.BookType == market {
+			if err = config.DB.Table("trading_account").Where("user_id=?", p.UserId).First(&account).Error; err != nil {
+				return err
+			}
+			account.Balance = account.Balance + int64((p.OrderPrice-res.AveragePrice)*res.Quantity)
+			account.UpdatedAt = time.Now()
+			if err = config.DB.Table("trading_account").Where("user_id=?", p.UserId).Updates(&account).Error; err != nil {
+				return err
+			}
+		} else {
+			if err = config.DB.Table("trading_account").Where("user_id=?", p.UserId).First(&account).Error; err != nil {
+				return err
+			}
+			account.Balance = account.Balance + int64((p.LimitPrice-res.AveragePrice)*res.Quantity)
```

Comment:
[Comment](https://github.com/RohanDudeja/Tradeasy/pull/39/files#r685698297)

Reply:
Its taking care of the left amount which can be added/subtracted.